perma_web/perma/forms.py

```diff
@@ -200,18 +248,44 @@ class UserAddVestingOrgForm(forms.ModelForm):
     """
 
     def __init__(self, *args, **kwargs):
-      registrar_id = False
-      if 'registrar_id' in kwargs:
-        registrar_id = kwargs.pop('registrar_id')
-      super(UserAddVestingOrgForm, self).__init__(*args, **kwargs)
-      if registrar_id:
-        self.fields['vesting_org'].queryset = VestingOrg.objects.filter(registrar_id=registrar_id).order_by('name')
-    
+        registrar_id = False
+        vesting_org_member_id = False
+        target_user_id = False
+
+        if 'registrar_id' in kwargs:
+            registrar_id = kwargs.pop('registrar_id')
+
+        if 'vesting_org_member_id' in kwargs:
+            vesting_org_member_id = kwargs.pop('vesting_org_member_id')
+
+        if 'target_user_id' in kwargs:
+            target_user_id = kwargs.pop('target_user_id')
+
+        super(UserAddVestingOrgForm, self).__init__(*args, **kwargs)
+
+        target_user = LinkUser.objects.get(pk=target_user_id)
+
+        # Vesting managers can only edit their own vesting members
+        if registrar_id:
+            # Get the union of the user's and the registrar member's vesting orgs
```

Comment:
This seems like getting the difference, not the union? I'm not clear what the 'exclude' is for.

Since it's the same in all three branches, maybe move it down after the if/else block, like

`vesting_orgs = vesting_orgs.exclude(pk__in=target_user.vesting_org.all())`

with a comment explaining what we're excluding?